src/zarr/array.py

```diff
@@ -62,7 +63,9 @@
 )
 from zarr.metadata import ArrayMetadata, ArrayV2Metadata, ArrayV3Metadata
 from zarr.store import StoreLike, StorePath, make_store_path
+from zarr.store.core import contains_array
 from zarr.sync import sync
+from zarr.v2.util import guess_chunks
```

Comment:
We probably shouldn't import from `zarr.v2` in v3 code.

Reply:
generally I agree, but the explicit goal here is v2 compatibility. we could re-implement the exact same code in the v3 codebase, but I think that has greater drawbacks.

Reply:
Didn't we decide to remove the `zarr.v2` modules before releasing 3.0?

Reply:
on the other hand, it would be good to get this routine in v3 so that we can make improvements to it. So I'm +1 on removing the v2 import and porting the logic. I will make that change.

Reply:
> Didn't we decide to remove the zarr.v2 modules before releasing 3.0?

yes, I think we did. that's another good reason to port this routine.

Reply:
[5abaab6](https://github.com/zarr-developers/zarr-python/pull/2042/commits/5abaab6b1f3f4409d40e14d31803ac0f41eaa2ab) brings `guess_chunks` over to v3, and adds tests.